### PR TITLE
evidently 0.3.3 :snowflake:

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://github.com/evidentlyai/evidently/releases
License: https://github.com/evidentlyai/evidently/blob/v0.3.3/LICENSE
Requirements: 
- https://github.com/evidentlyai/evidently/blob/v0.3.3/requirements.min.txt
- https://github.com/evidentlyai/evidently/blob/v0.3.3/setup.cfg
- https://github.com/evidentlyai/evidently/blob/v0.3.3/setup.py

Actions:
1. Clean up the recipe:
- Remove `noarch` python
- Skip `py<37`
- Skip `s390x` as `umap-learn` isn't available on s390x
- Add flags `--no-deps --no-build-isolation` to `script`
- Add missing `setuptools` & `wheel` to `host`
- Fix `python` in `host` and `run`
2. Update runtime dependencies: remove `dataclasses >=0.6`, fix pin for `pydantic >=1.9,<2`
3. Add `numba >=0.57.0` to `test/requires` to fix `SystemError: initialization of _internal failed without raising an exception` caused by an incorrect version of `numba` or `numpy` in the test environment, see also https://stackoverflow.com/questions/74947992/how-to-remove-the-error-systemerror-initialization-of-internal-failed-without
4. Add a test suite: `source_files`, `pytest` and the pytest command, to check if there can be issues we are not aware
5. Add `dev_url`
6. Add a `description`
7. Add `license_family`

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -37,16 +38,27 @@ requirements:
 test:
   imports:
     - evidently
-  commands:
-    - pip check
+  source_files:
+    - tests/*.py
```
Comment (rafaelmartins) [inline recipe/meta.yaml]: is this correct? there are subdirs with more tests in `tests/`, these tests are not required?

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Thanks. Fixed it. Trying to rebuild with all tests

Comment (skupr-anaconda) [reply recipe/meta.yaml]: As it fails now I'll debug it tomorrow. Some packages are missing for the test environment (`ipython`, etc.)

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Done!

**Review by rafaelmartins — APPROVED** [review]

**Review by sumit0190 — APPROVED** [review]

**Pull request merged**